networklibrary/src/main/java/com/eis/smsnetwork/SMSNetworkManager.java

```diff
@@ -11,18 +15,30 @@
 import com.eis.communication.network.listeners.InviteListener;
 import com.eis.communication.network.listeners.RemoveResourceListener;
 import com.eis.communication.network.listeners.SetResourceListener;
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSMessageHandler;
 import com.eis.smslibrary.SMSPeer;
+import com.eis.smsnetwork.broadcast.BroadcastReceiver;
+import com.eis.smsnetwork.smsnetcommands.SMSAcceptInvite;
+import com.eis.smsnetwork.smsnetcommands.SMSAddResource;
+import com.eis.smsnetwork.smsnetcommands.SMSInvitePeer;
+import com.eis.smsnetwork.smsnetcommands.SMSRemoveResource;
 
 /**
  * The manager class of the network.
  *
  * @author Edoardo Raimondi
  * @author Marco Cognolato
+ * @author Giovanni Velludo
  */
-public class SMSNetworkManager implements NetworkManager<String, String, SMSPeer, FailReason> {
+public class SMSNetworkManager implements NetworkManager<String, String, SMSPeer, SMSFailReason> {
 
     private NetSubscriberList<SMSPeer> netSubscribers = new SMSNetSubscriberList();
     private NetDictionary<String, String> netDictionary = new SMSNetDictionary();
+    private NetSubscriberList<SMSPeer> invitedPeers = new SMSNetSubscriberList();
```

Comment:
Why did you use a `NetSubscriberList`? Invited are not "subscribers" so it might cause some confusion, wasn't an ArrayList good enough?

Reply:
Yeah you're right, I'll change it 👍

Reply:
Where is the confusion? The name clearly says those are the invited peers. Using an ArrayList doesn't prevent duplicate peers being added to the list, while the Set we're using in NetSubscriberList prevents that. Using a HashSet also makes searches in the set perform better compared to an ArrayList